From d14db25ca6c9b701643eed545d6c7d494d168ce7 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Sat, 21 Apr 2018 05:36:43 +0200
Subject: [PATCH 1/3] klte-common: Stop abusing global contexts for fingerprint

* vcs_device is used to label /dev/vcs*, which are virtual consoles
* Create and use our own label for /dev/vfsspi so our fingerprint
  hal can access it, and rename vcs_data_file while we're at it

Change-Id: I01f0e8c4924d3847383319ce59dbbf802f89a36b
---
 sepolicy/common/device.te                  | 2 ++
 sepolicy/common/file.te                    | 2 +-
 sepolicy/common/file_contexts              | 4 ++--
 sepolicy/common/hal_fingerprint_default.te | 6 +++---
 sepolicy/common/tee.te                     | 4 ++--
 5 files changed, 10 insertions(+), 8 deletions(-)
 create mode 100644 sepolicy/common/device.te

diff --git a/sepolicy/common/device.te b/sepolicy/common/device.te
new file mode 100644
index 0000000..a3f6fe2
--- /dev/null
+++ b/sepolicy/common/device.te
@@ -0,0 +1,2 @@
+# Fingerprint
+type vfsspi_device, dev_type;
diff --git a/sepolicy/common/file.te b/sepolicy/common/file.te
index bf73383..9b03bb7 100644
--- a/sepolicy/common/file.te
+++ b/sepolicy/common/file.te
@@ -1 +1 @@
-type vcs_data_file, file_type, data_file_type;
+type vfsspi_data_file, file_type, data_file_type;
diff --git a/sepolicy/common/file_contexts b/sepolicy/common/file_contexts
index d2c1c11..eb7afe3 100644
--- a/sepolicy/common/file_contexts
+++ b/sepolicy/common/file_contexts
@@ -1,9 +1,9 @@
 # data files
-/data/validity(/.*)?                                    u:object_r:vcs_data_file:s0
+/data/validity(/.*)?                                    u:object_r:vfsspi_data_file:s0
 
 # device nodes
 /dev/ttyHS3                                             u:object_r:audio_device:s0
-/dev/vfsspi                                             u:object_r:vcs_device:s0
+/dev/vfsspi                                             u:object_r:vfsspi_device:s0
 
 # sysfs
 /sys/devices(/.*)?/input/input[1-2]/enabled             u:object_r:sysfs_hal_pwr:s0
diff --git a/sepolicy/common/hal_fingerprint_default.te b/sepolicy/common/hal_fingerprint_default.te
index 17bc016..8c06be7 100644
--- a/sepolicy/common/hal_fingerprint_default.te
+++ b/sepolicy/common/hal_fingerprint_default.te
@@ -1,6 +1,6 @@
 r_dir_file(hal_fingerprint_default, firmware_file)
 
 allow hal_fingerprint_default tee_device:chr_file rw_file_perms;
-allow hal_fingerprint_default vcs_data_file:dir rw_dir_perms;
-allow hal_fingerprint_default vcs_data_file:file create_file_perms;
-allow hal_fingerprint_default vcs_device:chr_file rw_file_perms;
+allow hal_fingerprint_default vfsspi_data_file:dir rw_dir_perms;
+allow hal_fingerprint_default vfsspi_data_file:file create_file_perms;
+allow hal_fingerprint_default vfsspi_device:chr_file rw_file_perms;
diff --git a/sepolicy/common/tee.te b/sepolicy/common/tee.te
index 02eeef6..ee97521 100644
--- a/sepolicy/common/tee.te
+++ b/sepolicy/common/tee.te
@@ -1,2 +1,2 @@
-allow tee vcs_data_file:dir create_dir_perms;
-allow tee vcs_data_file:file create_file_perms;
+allow tee vfsspi_data_file:dir create_dir_perms;
+allow tee vfsspi_data_file:file create_file_perms;
-- 
2.17.0

